api/users/views.py

```diff
@@ -662,3 +662,20 @@ def perform_destroy(self, instance):
             if val['id'] in current_institutions:
                 user.remove_institution(val['id'])
         user.save()
+
+class UserPublicFiles(JSONAPIBaseView, generics.ListAPIView, UserMixin, ODMFilterMixin):
+    view_name = 'user-public-files'
+    view_category = 'users'
+    serializer_class = PublicFilesSerializer
+
+    required_read_scopes = [CoreScopes.USERS_READ, CoreScopes.NODE_REGISTRATIONS_READ]
+    required_write_scopes = [CoreScopes.USERS_WRITE, CoreScopes.NODE_REGISTRATIONS_WRITE]
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    def get_queryset(self):
+
+        return Node.find(Q('contributors', 'eq', self.get_user()._id) & Q('is_public_files_collection', 'eq', True))
```

Comment:
So, this is for a *user*'s public *files*, but it returns a serialized node? It seems like it should behave more like the `/files/` endpoint than like the `/nodes/` endpoint.